lib/task.js

```diff
@@ -318,7 +339,11 @@ export default class Task {
     ));
 
     try {
-      await this.runtime.stats.timeGen(`tasks.time.states.${stat}.killed`, this.states.killed(this));
+      // Run killed hook and record reason why we aborted
+      await this.runtime.stats.timeGen(
```

Comment:
Nope... you have a stat for each reason it was aborted... that is not recorded by `_invoke`...

This changes the layout of your stats, but it'll now be `tasks.time.states.killed-by-abort.${stat}` in addition to ``tasks.time.states.killed`.

I believe the `stat` variable is some sort of reason... :)